AtelierTomato.Markov/AtelierTomato.Markov.Generation/GenerateMarkovSentence.cs

```diff
@@ -4,31 +4,21 @@
 
 namespace AtelierTomato.Markov.Generation
 {
-	public class GenerateMarkovSentence
+	public class GenerateMarkovSentence(ISentenceAccess sentenceAccess, IOptions<MarkovGenerationOptions> options)
 	{
-		private readonly ISentenceAccess sentenceAccess;
-		private readonly MarkovGenerationOptions options;
-		private static Random random = new Random();
+		private readonly ISentenceAccess sentenceAccess = sentenceAccess;
+		private readonly MarkovGenerationOptions options = options.Value;
+		private static readonly Random random = new();
 
-		public GenerateMarkovSentence(ISentenceAccess sentenceAccess, IOptions<MarkovGenerationOptions> options)
+		public async Task<string> Generate(SentenceFilter filter)
 		{
-			this.sentenceAccess = sentenceAccess;
-			this.options = options.Value;
-		}
-
-		public async Task<string> Generate()
-		{
-			Sentence? sentence = await GetFirstSentence();
-			if (sentence is null)
-			{
-				throw new Exception("Couldn't query any messages.");
-			}
+			Sentence? sentence = await GetFirstSentence(filter) ?? throw new Exception("Couldn't query any messages.");
 			string firstWord = sentence.Text.Substring(0, sentence.Text.IndexOf(' '));
 			var tokenizedSentence = new List<string> { firstWord };
 			var prevList = tokenizedSentence;
 
 			// Tracks the IDs of previously used sentences so that we don't recreate an existing sentence or ping pong between two sentences.
-			var prevIDs = new List<Guid> { sentence.ID };
+			var prevIDs = new List<string> { sentence.OID.ToString() };

```

Comment:
Can't we make this List\<IObjectOID\>?

Reply:
Will implement in SentenceAccess branch, not a change for this pull request.